src/base/emu-i386/simx86/codegen-sim.c

```diff
@@ -3019,6 +3020,23 @@ unsigned int CloseAndExec_sim(unsigned int PC, int mode)
 	return ret;
 }
 
+static void emu_pagefault_handler(dosaddr_t addr, int err, uint32_t op, int len)
+{
+	if (!in_emu_cpu()) {
+		default_sim_pagefault_handler(addr, err, op, len);
```

Comment:
Do we actually still pagefault
in sim? I have some memories
that you removed that at some
point?

Reply:
Indeed I removed that. This pagefault is simulated: there is a cache of unprotected pages, and if the page is outside the cache the write will check various things and call the simulated handler if the page is not accessible (for LDT, DPMI, self-modifying code; vgaemu and ROM are handled directly in the r/w functions).